File: python学习/python高级特性/1切片/1.py
L = ['fri', 'fridolph', 'yk', 'keke', 'k']

r = []
n = 3
for i in range(n):
  r.append(L[i])

# print(r)

# L[0:3]表示，从索引0开始，直到索引3为止，但不包括索引3
# print(L[0: 3])
# 若第一个是0，还可以省略
print(L[: 3])


L = list(range(100))

# 前10个
# print(L[:10])
# 后10个
# print(L[-10:])
# 前11-20个
# print(L[10:20])

# 前10个数，每两个取一个
# print(L[:10:2])

# 所有数，每5个取一个
# print(L[::5])

# 复制一个list
# print(L[:])

# tuple也是一种list，唯一区别是tuple不可变。因此，tuple也可以用切片操作，只是操作的结果仍是tuple
print((0,1,2,3,4,5)[:3])

File: 05Nodejs/Node-zns/08ejs/1.js
var ejs = require('ejs')
var fs = require('fs')

var str = '';

ejs.renderFile('./views/2.ejs', {
  title: '我是标题',
  name: 'fridolph',
  json: {
    arr: [
      { user: 'fri', pass: '111' },
      { user: 'yk', pass: '222' },
      { user: 'wb', pass: '333' }
    ]
  }
}, function(err,data) {
  if (err) { 
    // console.log('渲染错误');
    console.error(err);
  } else {
    console.log('渲染成功');
    console.log(data);

    str = data
  }
})

fs.writeFile('./www/index.html', str, function(err) {
  if (err) {
    console.log('写入错误');
    console.error(err);
  } else {
    console.log('写入成功');
  }
})